### PR TITLE
[feature/pipe] Implementing pipe validator 

### PR DESCRIPTION
**Pull request by oh-math**
Using pipe for validate the given name ocurrency in the database before its creation.

### CODE REVIEW

**Review by Joabesv — CHANGES_REQUESTED** [review]
Drop **joi** and favor **Zod**, since it's more suited for Typescript applications

File: src/user/user.service.ts [hunk 1]
```diff
@@ -7,6 +8,10 @@ import { CreateUserDto, UpdateUserDto, UserResponse } from './dto';
 export class UserService {
   constructor(private readonly prisma: PrismaService) {}
 
+  public async count(options?: Prisma.UserCountArgs): Promise<Number> {
+    return await this.prisma.user.count(options);
```
Comment (Joabesv) [inline src/user/user.service.ts:12]: Don't use `Number` as a type. Use number instead
you're calling the constructor, not the type here

File: src/user/pipe/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './count-name.pipe';
```
Comment (Joabesv) [inline src/user/pipe/index.ts:1]: Why use this pattern?

File: src/user/pipe/count-name.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {
```
Comment (Joabesv) [inline src/user/pipe/count-name.pipe.ts:1]: Drop **joi** and favor **Zod**

**Review by Joabesv — COMMENTED** [review]
Don't use `Number` as a type. Use number instead

File: src/user/pipe/index.ts (hunk 2 above)
Comment (oh-math) [reply src/user/pipe/index.ts:1]: You're talking about `barrels`?
This pattern allow you to export all items from a module with a single import, like that:
instead 
`
import { Foo } from './demo/foo.js'
 import { Bar } from './demo/bar.js'
` 
you can just do:
`import {Foo, Bar} from './demo'`

File: src/user/user.service.ts (hunk 1 above)
Comment (oh-math) [reply src/user/user.service.ts:12]: Great reminder, thx!

File: src/user/pipe/count-name.pipe.ts (hunk 3 above)
Comment (oh-math) [reply src/user/pipe/count-name.pipe.ts:1]: gonna open a issue fot that

File: src/user/pipe/index.ts (hunk 2 above)
Comment (Joabesv) [reply src/user/pipe/index.ts:1]: hmm, I used to think this was very ugly, LOL
but can see the use Cases of it

File: src/user/pipe/count-name.pipe.ts (hunk 3 above)
Comment (Joabesv) [reply src/user/pipe/count-name.pipe.ts:1]: mind if I take it?

Comment (oh-math) [reply src/user/pipe/count-name.pipe.ts:1]: nope, i wanna see that

**Review by Joabesv — APPROVED** [review]

**Comment by Joabesv** [comment]
LGTM

**Pull request merged**